### PR TITLE
Add Filesystem based standard test base

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Adds a base image for filesystem based standard test (i.e: where all the needed resources are provided from the file system). Also adds a gradle plugin to consume this. An example snippet of using this would be to add in `build.gradle`: 

```
airbyteStandardFsSourceTest { 
  configPath = "/path/to/config"
  specPath = "/path/to/spec"
  configuredCatalogPath = "/path/to/catalog"
} 
```
This was made as a supporting change for @gordalina 's elixir google analytics source 

## Reading Order
1. `airbyte-standard-fs-source-test.gradle `
1. `TestSourceMain.java`
1. `ExecutableTestSource.java`
1. anything else

### CODE REVIEW

File: airbyte-integrations/bases/base-standard-fs-source-test/src/main/java/io/airbyte/integrations/standardtest/source/fs/ExecutableTestSource.java [hunk 1]
```diff
@@ -22,13 +22,17 @@
  * SOFTWARE.
  */
 
-package io.airbyte.integrations.standardtest.source;
+package io.airbyte.integrations.standardtest.source.fs;
```
Comment (sherifnada) [inline airbyte-integrations/bases/base-standard-fs-source-test/src/main/java/io/airbyte/integrations/standardtest/source/fs/ExecutableTestSource.java:26]: this file was moved from standard-source-test. It was unused.

File: airbyte-integrations/bases/base-standard-fs-source-test/src/main/java/io/airbyte/integrations/standardtest/source/fs/TestSourceMain.java [hunk 2]
```diff
@@ -21,11 +21,11 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
+package io.airbyte.integrations.standardtest.source.fs;
```
Comment (sherifnada) [inline airbyte-integrations/bases/base-standard-fs-source-test/src/main/java/io/airbyte/integrations/standardtest/source/fs/TestSourceMain.java]: this file was moved from standard-source-test. It was unused.

**Review by jrhizor — APPROVED** [review]
Looks great. 

Two things:
1. The ordering of the name is really awkward. It isn't testing file system sources. `standardSourceTestFile` would be better and would match `standardSourceTestPython`. 
2. Outside of this PR, we should migrate any tests that don't have custom behavior to this which will save a lot of docker building in testing. Maybe even the template should use this one? Probably should just have a ticket for now.

**Comment by sherifnada** [comment]
Issue created: https://github.com/airbytehq/airbyte/issues/1334
Rename done

File: airbyte-integrations/bases/base-standard-source-test-file/src/main/java/io/airbyte/integrations/standardtest/source/fs/ExecutableTestSource.java [hunk 3]
```diff
@@ -101,7 +105,12 @@ protected ConfiguredAirbyteCatalog getConfiguredCatalog() {
 
   @Override
   protected JsonNode getState() {
-    return Jsons.deserialize(IOs.readFile(TEST_CONFIG.getStatePath()));
+    if (TEST_CONFIG.getStatePath() != null) {
+      return Jsons.deserialize(IOs.readFile(TEST_CONFIG.getStatePath()));
+    } else {
+      return Jsons.deserialize("{}}");
```
Comment (sherifnada) [inline airbyte-integrations/bases/base-standard-source-test-file/src/main/java/io/airbyte/integrations/standardtest/source/fs/ExecutableTestSource.java:111]:
```suggestion
      return Jsons.deserialize("{}");
```

**Pull request merged**